pkg/cache/cache.go

```diff
@@ -128,6 +128,18 @@ type Options struct {
 	// Be very careful with this, when enabled you must DeepCopy any object before mutating it,
 	// otherwise you will mutate the object in the cache.
 	UnsafeDisableDeepCopyByObject DisableDeepCopyByObject
+
+	// TransformFuncByObject is a map from GVKs to transformer functions which
+	// get applied when objects of the transformation are about to be committed
+	// to cache.
+	//
+	// This function is called both for new objects to enter the cache,
+	// 	and for updated objects.
+	TransformFuncByObject TransformFuncByObject
+
+	// DefaultTransform is the transform used for all GVKs which do
+	// not have an explicit transform func set in TransformFuncByObject
+	DefaultTransform toolscache.TransformFunc
```

Comment:
Nit: option fields should be matched. `TransformFuncByObject` with `DefaultTransformFunc`, or `TransformByObject` with `DefaultTransform`. (I prefer the lower ones)

Reply:
I've renamed the public-facing API to use the form without `Func`